cli/cmd/sync2.go

```diff
@@ -186,3 +189,16 @@ func getTablesForSpec(ctx context.Context, sourceClient *clients.SourceClient, s
 	}
 	return tables, true, err
 }
+
+// returns only the top-level tables in the given tables list, i.e. tables
+// with no parents
+func topLevelTables(allTables, tables schema.Tables) schema.Tables {
+	var top schema.Tables
+	for _, t := range tables {
+		if allTables.GetTopLevel(t.Name) == nil {
```

Comment:
Why do it this way, rather than checking if `t.Parent` is `nil`?

Reply:
Yeah, that was what I tried initially. Then I realized that the object we get back from `getTablesFromSpec` is deserialized from JSON, so it loses all the pointers that were present in the original object. For this reason it also isn't simple to set the parent pointers first and _then_ checking `t.Parent` for nil. I felt like this solution is the cleanest for now.

Reply:
We might need to reestablish those pointers/table references in a later release.